### PR TITLE
[wip] backupcontroller

### PR DESCRIPTION
**Pull request by hexfusion**
Signed-off-by: Sam Batschelet <sbatsche@redhat.com>

### CODE REVIEW

File: pkg/operator/backupcontroller/backupcontroller.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+	hostResourceDirDir := "/etc/kubernetes/static-pod-resources"
+	hostPodManifestDir := "/etc/kubernetes/manifests"
+
+	args := []string{
+		fmt.Sprintf("-v=%d", 2),
+		fmt.Sprintf("--revision=%d", 2),
+		fmt.Sprintf("--namespace=%s", "openshift-etcd"),
+		fmt.Sprintf("--pod=%s", "backup-pod"),
+		fmt.Sprintf("--resource-dir=%s", hostResourceDirDir),
+		fmt.Sprintf("--pod-manifest-dir=%s", hostPodManifestDir),
```
Comment (hexfusion) [inline pkg/operator/backupcontroller/backupcontroller.go:166]: look into dropping this as we don't need the manifest dir created by the installer.

File: pkg/operator/backupcontroller/backupcontroller.go [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+
+func (c *BackupConfigController) getSubstitutionReplacer(operatorSpec *operatorv1.StaticPodOperatorSpec, imagePullSpec, operatorImagePullSpec string) (*strings.Replacer, error) {
+	return strings.NewReplacer(
+		"${IMAGE}", imagePullSpec,
+		"${KS_INSTALLER_FLAGS}", strings.Join(getKSArgs(), "\n"),
+		"${KAS_INSTALLER_FLAGS}", strings.Join(getKASArgs(), "\n"),
+		"${KCM_INSTALLER_FLAGS}", strings.Join(getKCMArgs(), "\n"),
+		"${ETCD_INSTALLER_FLAGS}", strings.Join(getEtcdArgs(), "\n"),
+	), nil
+}
```
Comment (ironcladlou) [inline pkg/operator/backupcontroller/backupcontroller.go:172]: All these manipulations seem to be making things unnecessarily difficult and brittle — what if instead we deserialize the bindata as an API object, do manipulations against the actual Go type, then re-serialize the result?

Comment (hexfusion) [reply pkg/operator/backupcontroller/backupcontroller.go]: yeah this is very WIP I agree we can just inject on `env`

**Comment by hexfusion** [comment]
Man CI is odd not sure why this closed..

**Pull request closed without merging**